src/compiler/types.ts

```diff
@@ -7003,16 +7004,17 @@ namespace ts {
         ContainsTypeScript = 1 << 0,
         ContainsJsx = 1 << 1,
         ContainsESNext = 1 << 2,
-        ContainsES2022 = 1 << 3,
-        ContainsES2021 = 1 << 4,
-        ContainsES2020 = 1 << 5,
-        ContainsES2019 = 1 << 6,
-        ContainsES2018 = 1 << 7,
-        ContainsES2017 = 1 << 8,
-        ContainsES2016 = 1 << 9,
-        ContainsES2015 = 1 << 10,
-        ContainsGenerator = 1 << 11,
-        ContainsDestructuringAssignment = 1 << 12,
+        ContainsES2023 = 1 << 3,
+        ContainsES2022 = 1 << 4,
+        ContainsES2021 = 1 << 5,
+        ContainsES2020 = 1 << 6,
+        ContainsES2019 = 1 << 7,
+        ContainsES2018 = 1 << 8,
+        ContainsES2017 = 1 << 9,
+        ContainsES2016 = 1 << 10,
+        ContainsES2015 = 1 << 11,
+        ContainsGenerator = 1 << 12,
+        ContainsDestructuringAssignment = 1 << 13,

```

Comment:
this now overlaps with `ContainsTypeScriptClassSyntax = 1 << 13`, which means we're officially out of space in TransformFlags. =(

Paging @rbuckton for advice.